fbgemm_gpu/include/fbgemm_gpu/sparse_ops_utils.h

```diff
@@ -216,6 +216,14 @@ inline bool torch_tensor_empty_or_on_cpu_check(
       " and ",                                                         \
       (y).numel())
 
+#define TENSORS_HAVE_SAME_SYM_NUMEL(x, y)                              \
+  TORCH_CHECK(                                                         \
+      (x).sym_numel() == (y).sym_numel(),                              \
+      #x " must have the same number of elements as " #y " They had ", \
+      (x).sym_numel(),                                                 \
+      " and ",                                                         \
+      (y).sym_numel())
```

Comment:
A more robust version of this which works better with unbacked symints is

```
TORCH_SYM_CHECK(x.sym_numel().sym_eq(y.sym_numel()))
```

for more explanation on what this is doing, see https://docs.google.com/document/d/1HSuTTVvYH1pTew89Rtpeu84Ht3nQEFTYhAX3Ypa_xJs/edit#heading=h.jqnrfurlygn5